### ISSUE TITLE
Latest tag in dockerhub does not appear to be 3.4.1

### ISSUE THREAD

**Issue by DarklyAutumnClad** [openstf/stf#1207]
I pulled latest and v3.4.1, what ever you have in latest in dockerhub is not v3.4.1 https://hub.docker.com/r/openstf/stf/tags

The sha256 digests do not match, although, what is tagged as latest returns 3.4.1 when stf --version is called.
The directly tagged v3.4.1, when pulled and started, does not require me to specify additional MQ for the client/device side.

That release, as of now, at 3 days old.  

This conflicts with the official latest release which is 3.4.1, even though the build reports that as 3.4.1

As I mentioned, it appears this adds mq requirements for the client side, which my test farm was not prepared for when pulling latest. 

No changes to documentation either to show said requirements, though I suppose not, since this doesn't match the official release. Fails out trying to come up with ports for a client STF MQ setup.

Some good news though, I discovered during the mismatch with api/processor/triproxies/auth on 3.4.1 and the provider on whatever version you have there in :latest, and suddenly android 10 devices are coming online. So, a silver lining to look forward to. Previously, using the v3.4.1 tag did not work with that, so whatever sauce you've baked into what ever actual version of stf is tagged latest, is a positive. 

To reiterate:

:latest requires client mq connections and reports 3.4.1
while
v3.4.1 tag does not require mq connections for client side, and reports 3.4.1
What's more:
v3.4.1 when used for provider node, does not produce working android 10 connections.
:latest, which again, reports as v3.4.1 does produce working connections for devices running andoid 10.

**Comment by koral--** [comment]
Latest tag on Docker Hub is autobuilt from the current snapshot on master.
Indeed `stf --version` should not return the same output as for 3.4.1 tag.

There were updates adding Android 10 support. You should update all of the units to the same version.

It is not yet decided what will be the next version number and what is its exact scope.

**Comment by DarklyAutumnClad** [comment]
Yes, that is my desire in reporting this. These additional ZeroMQ updates, they appear to be required for the device side now as well, at least by the version tagged latest, the API cannot boot properly without those two new options:

**api_1                   | Missing required arguments: connect-push-dev, connect-sub-dev**

`api_1                   |   --connect-push-dev, --pd  Device-side ZeroMQ PULL endpoint to connect to.
api_1                   |                                                               [array] [required]
api_1                   |   --connect-sub-dev, --sd   Device-side ZeroMQ PUB endpoint to connect to.
api_1                   |                                                               [array] [required]`

So am I correct in assuming:

That I simply need to inspect the options for stf provider, and for api, and add them to my docker's command section, and make sure the ports are good to go? 

Thanks.

**Comment by denis99999** [comment]
@DarklyAutumnClad 

New options namely `--connect-push-dev` and `--connect-sub-dev` are required for the legacy `api`
unit and the new `groups-engine` unit as described in [DEPLOYMENT.md](../tree/master/doc/DEPLOYMENT.md) documentation, and both units are part of the application side (i.e. not part of the device side), so there is nothing to change at device side in particular for `provider` service.

**Comment by DarklyAutumnClad** [comment]
@denis99999 Thanks for that, I had not realized this had changed, I've only followed the project as more an admin that set it up some moons ago. I am glad I stuck around for the day here, I will give that backend change a shot there, since I have both image versions pulled locally.

**Comment by DarklyAutumnClad** [comment]
I am just stating where I am for posterity, nothing to followup for on this comment (this bug mainly concerns the versioning that they have for docker images, but for anyone who comes later, or is linked via search engine, related to additional keywords: 

I am working again with some version beyond 3.4.1,  by adding/using the -connect-sub-dev --connect-push-dev values set to tcp://dev-triproxy:72xx on the api, using :latest, which is some version post 3.4.1
